File: projects/common/interface.py
import sys, datetime


def usage():
    print('usage: select_date_HHMM(20240619_1610) output_filename')
    print('')
    sys.exit(-1)


def chk_para(inp_para):
    if len(inp_para) < 3:
        usage()
    par1 = inp_para[1]
    try:
        dt_obj = datetime.datetime.strptime(par1, "%Y%m%d_%H%M")
    except:
        usage()
    if dt_obj.year < 2000:
        usage()
    if len(inp_para[2]) < 3:
        usage()

File: projects/common/scala/src/main/scala/interface.scala
import java.time.format.DateTimeFormatter
import org.apache.spark.sql.SparkSession

object Interface {
 def usage(): Unit = {
  println("usage: select_date_HHMM(20240620_0000) output_filename nhours");
  val spark = SparkSession.builder.getOrCreate();
  spark.stop();
  sys.exit(-1);
 }

 def chk_para(args: Array[String]): Boolean = {
  val fmt = DateTimeFormatter.ofPattern("yyyyMMdd_HHmm");
  if(args.length < 3 || args(1).length < 3 || args(2).length < 3) false
  else {
   try {
    val dt = fmt.parse(args(1))
    val yf = DateTimeFormatter.ofPattern("yyyy")
    if (yf.format(dt).toInt < 2000) false else true
   } catch {case _: Throwable => false}
  }
 }
}
